### ISSUE TITLE
Hotkey save crash

### ISSUE THREAD

**Issue by nicolasdejong** [mzomparelli/zVirtualDesktop#103]
After upgrading had to set my wallpapers and hotkeys again.

Setting a hotkey action caused a crash:

![image](https://user-images.githubusercontent.com/12181216/30813825-ca9339ec-a20e-11e7-8721-c2d5ae2fd15b.png)

(ALT should have been CTRL, worked ok when I did that)

**Comment by mzomparelli** [comment]
I have confirmed this bug. This will be fixed in 1.0.68.

**Comment by mzomparelli** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Save Hotkey Bug/Crash](https://trello.com/c/lmvfmFpc/23-save-hotkey-bug-crash)

**Comment by mzomparelli** [comment]
This will crash every time if you replace a hotkey and I'm surprised no one mentioned it yet. Technically it should not have even compiled. I have corrected this bug and published a new version of 1.0.67. Anyone can download a new binary or wait for 1.0.68 to be released in a few days.

This would not occur if you edited the original. What happened here is that you tried to create a hotkey that already exists and it is replacing it. I've added a  message box here so you are alerted when this happens and you have the option to say no to replacing.

https://zomp.co/files.aspx?id=zvd

`var i = GLOBALS.Hotkeys.FindIndex(l => l.HotkeyString() == hk.HotkeyString());`

The above line caused the bug. `hk` doesn't even exist. It should be `hki`. How could this have even compiled if `hk` doesn't exist? Maybe because it's a lambda expression?

EDIT:
My mistake, `hk` does exist as an unused class wide variable. I've deleted that now.
